cirq-core/cirq/ops/three_qubit_gates_test.py

```diff
@@ -319,3 +319,24 @@ def test_resolve(resolve_fn):
     diagonal_gate = resolve_fn(diagonal_gate, {'b': 19})
     assert diagonal_gate == cirq.ThreeQubitDiagonalGate(diagonal_angles)
     assert not cirq.is_parameterized(diagonal_gate)
+
+
+def test_controlled_ops_consistency():
+    a, b, c, d = cirq.LineQubit.range(4)
+    assert cirq.CCX(a, b, c).controlled_by(d) == cirq.CCX(d, b, c).controlled_by(a)
+    assert cirq.CCZ(a, b, c).controlled_by(d) == cirq.CCZ(d, b, c).controlled_by(a)
+    assert cirq.CSWAP(a, b, c).controlled_by(d) == cirq.CSWAP(d, b, c).controlled_by(a)
+
+
+@pytest.mark.parametrize(
+    'gate',
+    [
+        cirq.CCX,
+        cirq.CCZ,
+        cirq.CSWAP,
+        cirq.CCXPowGate(global_shift=-0.5, exponent=1),
+        cirq.CCZPowGate(global_shift=-0.5, exponent=1),
+    ],
+)
+def test_controlled_gate_op_consistent(gate):
+    cirq.testing.assert_controlled_and_controlled_by_identical(gate)
```

Comment:
Could this be part of the standard gate test battery, instead of being performed special for these specific gates?

Reply:
Can you elaborate a bit on what you mean by the standard gate test battery?

Reply:
Was going to comment the same thing. This could be a part of `assert_implements_consistent_protocols`.